--- 0/win32_threads.c	2003-07-23 13:43:46.000000000 +0400
+++ win32_threads.c	2003-07-23 13:39:27.000000000 +0400
@@ -630,7 +630,7 @@
   LOCK();
   for (i = 0;
        i < MAX_THREADS &&
-       !thread_table[i].in_use || thread_table[i].id != thread_id;
+       (!thread_table[i].in_use || thread_table[i].id != thread_id);
        i++) {}
   if (i >= MAX_THREADS ) {
     WARN("thread %ld not found on detach", (GC_word)thread_id);
