*** ../0/gdome2-0.7.4/libgdome/gdome.c	Thu May 23 19:46:17 2002
--- ./libgdome/gdome.c	Fri Apr 11 13:09:18 2003
***************
*** 84,90 ****
   * Returns: the new DOMString object.
   */
  GdomeDOMString *
! gdome_str_mkref_xml (xmlChar *str)
  {
  	return gdome_xml_str_mkref_xml (str);
  }
--- 84,90 ----
   * Returns: the new DOMString object.
   */
  GdomeDOMString *
! gdome_str_mkref_xml (const xmlChar *str)
  {
  	return gdome_xml_str_mkref_xml (str);
  }
*** ../0/gdome2-0.7.4/libgdome/gdome.h	Thu May 23 19:46:18 2002
--- ./libgdome/gdome.h	Fri Apr 11 13:15:28 2003
***************
*** 254,259 ****
--- 254,260 ----
  GdomeDOMString     *gdome_str_mkref           (const gchar *str);
  GdomeDOMString     *gdome_str_mkref_own       (gchar *str);
  GdomeDOMString     *gdome_str_mkref_dup       (const gchar *str);
+ GdomeDOMString     *gdome_str_mkref_xml       (const unsigned char /*xmlChar*/ *str);
  void                gdome_str_ref             (GdomeDOMString *self);
  void                gdome_str_unref           (GdomeDOMString *self);
  GdomeBoolean        gdome_str_equal           (GdomeDOMString *self,
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-attribute.c	Wed Apr  3 11:05:00 2002
--- ./libgdome/gdomecore/gdome-xml-attribute.c	Fri Apr 11 12:12:42 2003
***************
*** 247,255 ****
    value = xmlNodeGetContent ((xmlNode *)priv->n);
  
    if (value != NULL)
!     return gdome_xml_str_mkref_own (value);
    else
!     return gdome_xml_str_mkref_dup("");
  }
  
  /**
--- 247,255 ----
    value = xmlNodeGetContent ((xmlNode *)priv->n);
  
    if (value != NULL)
!     return gdome_xml_str_mkref_xml (value);
    else
!     return gdome_xml_str_mkref ("");
  }
  
  /**
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-cdata.c	Mon Apr  1 21:08:24 2002
--- ./libgdome/gdomecore/gdome-xml-cdata.c	Fri Apr 11 12:12:42 2003
***************
*** 184,190 ****
  	g_return_val_if_fail (GDOME_XML_IS_CD (priv), NULL);
  	g_return_val_if_fail (exc != NULL, NULL);
  
! 	return gdome_xml_str_mkref_own (xmlNodeGetContent (priv->n));
  }
  
  /**
--- 184,190 ----
  	g_return_val_if_fail (GDOME_XML_IS_CD (priv), NULL);
  	g_return_val_if_fail (exc != NULL, NULL);
  
! 	return gdome_xml_str_mkref_xml (xmlNodeGetContent (priv->n));
  }
  
  /**
***************
*** 310,316 ****
  	ret[stop - start] = '\0';
  
  	xmlFree (str);
! 	return gdome_xml_str_mkref_own (ret);
  }
  
  /**
--- 310,316 ----
  	ret[stop - start] = '\0';
  
  	xmlFree (str);
! 	return gdome_xml_str_mkref_xml (ret);
  }
  
  /**
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-element.c	Mon Apr  1 21:08:24 2002
--- ./libgdome/gdomecore/gdome-xml-element.c	Fri Apr 11 13:12:35 2003
***************
*** 144,150 ****
  																									 (gchar *) gdome_xmlGetNsPrefix(priv->n),
  																									 (gchar *) gdome_xmlGetName(priv->n)));
  	else
! 		ret = gdome_xml_str_mkref_dup((gchar *)gdome_xmlGetName (priv->n));
  
  	return ret;
  }
--- 144,150 ----
  																									 (gchar *) gdome_xmlGetNsPrefix(priv->n),
  																									 (gchar *) gdome_xmlGetName(priv->n)));
  	else
! 		ret = gdome_xml_str_mkref_dup (gdome_xmlGetName (priv->n));
  
  	return ret;
  }
***************
*** 196,202 ****
  	value = xmlGetProp (priv->n, name->str);
  
  	if (value != NULL)
! 		return gdome_xml_str_mkref_own (value);
  	else {
  		strs = g_strsplit((gchar *)name->str, ":", 0);
  		if (xmlStrEqual(strs[0], "xmlns"))
--- 196,202 ----
  	value = xmlGetProp (priv->n, name->str);
  
  	if (value != NULL)
! 		return gdome_xml_str_mkref_xml (value);
  	else {
  		strs = g_strsplit((gchar *)name->str, ":", 0);
  		if (xmlStrEqual(strs[0], "xmlns"))
***************
*** 203,211 ****
  			ns = gdome_xmlGetNsDecl(priv->n, strs[1]);
  		g_strfreev(strs);
  		if (ns != NULL)
! 			return gdome_xml_str_mkref_dup ((gchar *)ns->href);
  		else
! 			return gdome_xml_str_mkref_dup ("");
  	}
  }
  
--- 203,211 ----
  			ns = gdome_xmlGetNsDecl(priv->n, strs[1]);
  		g_strfreev(strs);
  		if (ns != NULL)
! 			return gdome_xml_str_mkref_dup (ns->href);
  		else
! 			return gdome_xml_str_mkref ("");
  	}
  }
  
***************
*** 247,253 ****
  		return;
  	}
  
!   prevValue = gdome_xml_str_mkref_own (xmlGetProp (priv->n, name->str));
    attr = gdome_xml_n_mkref ((xmlNode *)xmlSetProp (priv->n, name->str, value->str));
  
    if (attr != NULL) {
--- 247,253 ----
  		return;
  	}
  
!   prevValue = gdome_xml_str_mkref_xml (xmlGetProp (priv->n, name->str));
    attr = gdome_xml_n_mkref ((xmlNode *)xmlSetProp (priv->n, name->str, value->str));
  
    if (attr != NULL) {
***************
*** 534,548 ****
  	value = xmlGetNsProp (priv->n, localName->str, namespaceURI->str);
  
  	if (value != NULL)
! 		return gdome_xml_str_mkref_own (value);
  	else if (xmlStrEqual(namespaceURI->str, GDOME_XMLNS_NAMESPACE)) {
  		ns = gdome_xmlGetNsDecl(priv->n, localName->str);
  		if (ns != NULL)
! 			return gdome_xml_str_mkref_dup ((gchar *)ns->href);
  		else
! 			return gdome_xml_str_mkref_dup ("");
  	} else {
! 		return gdome_xml_str_mkref_dup ("");
  	}
  }
  
--- 534,548 ----
  	value = xmlGetNsProp (priv->n, localName->str, namespaceURI->str);
  
  	if (value != NULL)
! 		return gdome_xml_str_mkref_xml (value);
  	else if (xmlStrEqual(namespaceURI->str, GDOME_XMLNS_NAMESPACE)) {
  		ns = gdome_xmlGetNsDecl(priv->n, localName->str);
  		if (ns != NULL)
! 			return gdome_xml_str_mkref_dup (ns->href);
  		else
! 			return gdome_xml_str_mkref ("");
  	} else {
! 		return gdome_xml_str_mkref ("");
  	}
  }
  
***************
*** 637,643 ****
  			g_free(localName);
  		}
  		else {
! 		 	prevValue = gdome_xml_str_mkref_own (xmlGetProp (priv->n,
  		 																									 (xmlChar *)qualifiedName->str));
  		 	attr = gdome_xml_n_mkref ((xmlNode *)xmlSetProp (priv->n,
  		 																									 (xmlChar *)qualifiedName->str,
--- 637,643 ----
  			g_free(localName);
  		}
  		else {
! 		 	prevValue = gdome_xml_str_mkref_xml (xmlGetProp (priv->n,
  		 																									 (xmlChar *)qualifiedName->str));
  		 	attr = gdome_xml_n_mkref ((xmlNode *)xmlSetProp (priv->n,
  		 																									 (xmlChar *)qualifiedName->str,
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-node.c	Wed Jan 22 00:29:16 2003
--- ./libgdome/gdomecore/gdome-xml-node.c	Fri Apr 11 12:12:42 2003
***************
*** 251,259 ****
    switch (type) {
    case XML_DOCUMENT_TYPE_NODE:
    case XML_DTD_NODE:
!     return gdome_xml_dt_unref (self, exc);
    case XML_DOCUMENT_NODE:
!     return gdome_xml_doc_unref (self, exc);
    default:
      if (priv->refcnt > 0)
        priv->refcnt--;
--- 251,261 ----
    switch (type) {
    case XML_DOCUMENT_TYPE_NODE:
    case XML_DTD_NODE:
!     gdome_xml_dt_unref (self, exc);
!     return;
    case XML_DOCUMENT_NODE:
!     gdome_xml_doc_unref (self, exc);
!     return;
    default:
      if (priv->refcnt > 0)
        priv->refcnt--;
***************
*** 634,640 ****
  	case XML_CDATA_SECTION_NODE:
  	case XML_COMMENT_NODE:
  	case XML_TEXT_NODE:
! 		ret = gdome_xml_str_mkref_own (xmlNodeGetContent (priv->n));
  		break;
  	case XML_ELEMENT_NODE:
  	case XML_ENTITY_NODE:
--- 636,642 ----
  	case XML_CDATA_SECTION_NODE:
  	case XML_COMMENT_NODE:
  	case XML_TEXT_NODE:
! 		ret = gdome_xml_str_mkref_xml (xmlNodeGetContent (priv->n));
  		break;
  	case XML_ELEMENT_NODE:
  	case XML_ENTITY_NODE:
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-pi.c	Mon Apr  1 21:08:24 2002
--- ./libgdome/gdomecore/gdome-xml-pi.c	Fri Apr 11 12:12:42 2003
***************
*** 144,150 ****
    g_return_val_if_fail (GDOME_XML_IS_PI (priv), NULL);
    g_return_val_if_fail (exc != NULL, NULL);
  
!  	return gdome_xml_str_mkref_own (xmlNodeGetContent (priv->n));
  }
  
  /**
--- 144,150 ----
    g_return_val_if_fail (GDOME_XML_IS_PI (priv), NULL);
    g_return_val_if_fail (exc != NULL, NULL);
  
!  	return gdome_xml_str_mkref_xml (xmlNodeGetContent (priv->n));
  }
  
  /**
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-str.c	Mon Apr  1 21:08:24 2002
--- ./libgdome/gdomecore/gdome-xml-str.c	Fri Apr 11 12:12:42 2003
***************
*** 51,57 ****
  
  void
  gdome_xml_str_unref_xml (GdomeDOMString *self) {
!   xmlFree (self->str);
  #ifdef DEBUG_REFCNT
        gdome_refdbg_delRef ((void *)self, GDOME_REFDBG_STR);
  #endif
--- 51,57 ----
  
  void
  gdome_xml_str_unref_xml (GdomeDOMString *self) {
!   xmlFree ((xmlChar*)self->str);
  #ifdef DEBUG_REFCNT
        gdome_refdbg_delRef ((void *)self, GDOME_REFDBG_STR);
  #endif
***************
*** 115,121 ****
   * Returns: the new DOMString object.
   */
  GdomeDOMString *
! gdome_xml_str_mkref_xml (xmlChar *str) {
    GdomeDOMString *ds;
  
    if (str == NULL)
--- 115,121 ----
   * Returns: the new DOMString object.
   */
  GdomeDOMString *
! gdome_xml_str_mkref_xml (const xmlChar *str) {
    GdomeDOMString *ds;
  
    if (str == NULL)
***************
*** 127,133 ****
  #endif
    ds->refcnt = 1;
  	ds->unref = gdome_xml_str_unref_xml;
! 	ds->str = str;
  	return ds;
  }
  
--- 127,133 ----
  #endif
    ds->refcnt = 1;
  	ds->unref = gdome_xml_str_unref_xml;
! 	ds->str = (gchar*)str;
  	return ds;
  }
  
*** ../0/gdome2-0.7.4/libgdome/gdomecore/gdome-xml-str.h	Tue Aug  7 01:56:08 2001
--- ./libgdome/gdomecore/gdome-xml-str.h	Fri Apr 11 12:12:42 2003
***************
*** 26,32 ****
  
  GdomeDOMString     *gdome_xml_str_mkref           (const gchar *str);
  GdomeDOMString     *gdome_xml_str_mkref_own       (gchar *str);
! GdomeDOMString     *gdome_xml_str_mkref_xml       (xmlChar *str);
  GdomeDOMString     *gdome_xml_str_mkref_dup       (const gchar *str);
  
  void                gdome_xml_str_ref             (GdomeDOMString *self);
--- 26,32 ----
  
  GdomeDOMString     *gdome_xml_str_mkref           (const gchar *str);
  GdomeDOMString     *gdome_xml_str_mkref_own       (gchar *str);
! GdomeDOMString     *gdome_xml_str_mkref_xml       (const xmlChar *str);
  GdomeDOMString     *gdome_xml_str_mkref_dup       (const gchar *str);
  
  void                gdome_xml_str_ref             (GdomeDOMString *self);
*** ../0/gdome2-0.7.4/libgdome/xpath/gdome-xpath-xpnsresolv.c	Fri Mar  1 06:33:47 2002
--- ./libgdome/xpath/gdome-xpath-xpnsresolv.c	Fri Apr 11 12:12:42 2003
***************
*** 114,120 ****
  
    ns = xmlSearchNs(priv->n->doc, priv->n, prefix->str);
    if (ns != NULL) {
!     return gdome_str_mkref_own ((gchar *)ns->href);
    } else {
      return NULL;
    }
--- 114,120 ----
  
    ns = xmlSearchNs(priv->n->doc, priv->n, prefix->str);
    if (ns != NULL) {
!     return gdome_str_mkref_xml (ns->href);
    } else {
      return NULL;
    }
