--- libgdome/gdomecore/gdome-xml-xmlmemory.c.0  Mon Nov 19 11:26:22 2001
+++ libgdome/gdomecore/gdome-xml-xmlmemory.c    Fri Feb 22 18:45:31 2002
@@ -33,10 +33,10 @@
 #include <parser.h>
 #include "gdome-xml-node.h"

-const xmlChar xmlStringText[] = { 't', 'e', 'x', 't', 0 };
-const xmlChar xmlStringTextNoenc[] =
+static const xmlChar xmlStringText[] = { 't', 'e', 'x', 't', 0 };
+static const xmlChar xmlStringTextNoenc[] =
   { 't', 'e', 'x', 't', 'n', 'o', 'e', 'n', 'c', 0 };
-const xmlChar xmlStringComment[] = { 'c', 'o', 'm', 'm', 'e', 'n', 't', 0 };
+const static xmlChar xmlStringComment[] = { 'c', 'o', 'm', 'm', 'e', 'n', 't', 0 };

 void gdome_xmlFreeProp (xmlAttrPtr cur);
 void gdome_xmlFreePropList (xmlAttrPtr cur);
