--- HTMLparser.c.0	Sat Nov 10 14:57:32 2001
+++ HTMLparser.c	Fri Dec 28 18:31:14 2001
@@ -1412,7 +1412,8 @@
  */
 int
 UTF8ToHtml(unsigned char* out, int *outlen,
-              const unsigned char* in, int *inlen) {
+              const unsigned char* in, int *inlen,
+			  void *info) {
     const unsigned char* processed = in;
     const unsigned char* outend;
     const unsigned char* outstart = out;
