--- error.c.0	Tue Oct 30 06:35:06 2001
+++ error.c	Fri Jan 25 13:28:18 2002
@@ -144,6 +144,7 @@
     int n;
     xmlChar  content[81];
     xmlChar *ctnt;
+    xmlChar curCh;
 
     if (input == NULL) return;
     cur = input->cur;
@@ -161,7 +162,7 @@
     n = 0;
     ctnt = content;
     while ((*cur != 0) && (*cur != '\n') && (*cur != '\r') && (n < 79)) {
-		*ctnt++ = *cur++;
+		*ctnt++ = ((curCh=*cur++)=='\t')?' ':curCh;
 	n++;
     }
     *ctnt = 0;
