--- libgdome/gdomecore/gdome-xml-node.h.0       Wed Jan 30 14:03:55 2002
+++ libgdome/gdomecore/gdome-xml-node.h Wed Jan 30 14:05:15 2002
@@ -106,7 +106,7 @@
        Gdome_xml_ListenerList *ll;
 };

-const GdomeNodeVtab gdome_xml_n_vtab;
+extern const GdomeNodeVtab gdome_xml_n_vtab;

 xmlNode           * gdome_xml_n_get_xmlNode      (GdomeNode *node);
 GdomeNode         * gdome_xml_n_mkref            (xmlNode *n);
@@ -201,6 +201,6 @@
                                                   GdomeNode *newChild,
                                                   GdomeException *exc);

-const GdomeNodeVtab gdome_xml_n_vtab;
+/*const GdomeNodeVtab gdome_xml_n_vtab;*/

 #endif /* GDOME_NODE_FILE */
